src/battle_anim_effects_2.c

```diff
@@ -0,0 +1,3867 @@
+    AFFINEANIMCMD_END,
+};
+
+const union AffineAnimCmd *const gUnknown_83E3B78[] =
+{
+    gUnknown_83E3B60,
+};
+
+// Unused
+const struct SpriteTemplate gUnknown_83E3B7C =
+{
+    .tileTag = ANIM_TAG_EXPLOSION_6,
+    .paletteTag = ANIM_TAG_EXPLOSION_6,
+    .oam = &gOamData_AffineNormal_ObjNormal_32x32,
+    .anims = gUnknown_83E3B5C,
+    .images = NULL,
+    .affineAnims = gUnknown_83E3B78,
+    .callback = AnimSpriteOnMonPos,
+};
+
+const union AnimCmd gKinesisZapEnergyAnimCmds[] =
+{
+    ANIMCMD_FRAME(0, 3, .hFlip = TRUE),
+    ANIMCMD_FRAME(8, 3, .hFlip = TRUE),
+    ANIMCMD_FRAME(16, 3, .hFlip = TRUE),
+    ANIMCMD_FRAME(24, 3, .hFlip = TRUE),
+    ANIMCMD_FRAME(32, 3, .hFlip = TRUE),
+    ANIMCMD_FRAME(40, 3, .hFlip = TRUE),
+    ANIMCMD_FRAME(48, 3, .hFlip = TRUE),
+    ANIMCMD_LOOP(1),
+    ANIMCMD_END,
+};
+
+const union AnimCmd *const gKinesisZapEnergyAnimTable[] =
+{
+    gKinesisZapEnergyAnimCmds,
+};
+
+const struct SpriteTemplate gKinesisZapEnergySpriteTemplate =
+{
+    .tileTag = ANIM_TAG_ALERT,
+    .paletteTag = ANIM_TAG_ALERT,
+    .oam = &gOamData_AffineOff_ObjNormal_32x16,
+    .anims = gKinesisZapEnergyAnimTable,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = Anim_KinesisZapEnergy,
+};
+
+const union AffineAnimCmd gSwordsDanceBladeAffineAnimCmds[] =
+{
+    AFFINEANIMCMD_FRAME(0x10, 0x100, 0, 0),
+    AFFINEANIMCMD_FRAME(0x14, 0x0, 0, 12),
+    AFFINEANIMCMD_FRAME(0x0, 0x0, 0, 32),
+    AFFINEANIMCMD_END,
+};
+
+const union AffineAnimCmd *const gSwordsDanceBladeAffineAnimTable[] =
+{
+    gSwordsDanceBladeAffineAnimCmds,
+};
+
+const struct SpriteTemplate gSwordsDanceBladeSpriteTemplate =
+{
+    .tileTag = ANIM_TAG_SWORD,
+    .paletteTag = ANIM_TAG_SWORD,
+    .oam = &gOamData_AffineNormal_ObjBlend_32x64,
+    .anims = gDummySpriteAnimTable,
+    .images = NULL,
+    .affineAnims = gSwordsDanceBladeAffineAnimTable,
+    .callback = Anim_SwordsDanceBlade,
+};
+
+const struct SpriteTemplate gSonicBoomSpriteTemplate =
+{
+    .tileTag = ANIM_TAG_AIR_WAVE,
+    .paletteTag = ANIM_TAG_AIR_WAVE,
+    .oam = &gOamData_AffineDouble_ObjBlend_32x16,
+    .anims = gDummySpriteAnimTable,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = AnimSonicBoomProjectile,
+};
+
+const struct SpriteTemplate gAirWaveProjectileSpriteTemplate =
+{
+    .tileTag = ANIM_TAG_AIR_WAVE,
+    .paletteTag = ANIM_TAG_AIR_WAVE,
+    .oam = &gOamData_AffineOff_ObjBlend_32x16,
+    .anims = gDummySpriteAnimTable,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = AnimAirWaveProjectile,
+};
+
+const union AffineAnimCmd gGrowingRingAffineAnimCmds[] =
+{
+    AFFINEANIMCMD_FRAME(0x20, 0x20, 0, 0),
+    AFFINEANIMCMD_FRAME(0x7, 0x7, 0, -56),
+    AFFINEANIMCMD_END,
+};
+
+const union AffineAnimCmd gWaterPulseRingAffineAnimCmds[] =
+{
+    AFFINEANIMCMD_FRAME(0x5, 0x5, 0, 10),
+    AFFINEANIMCMD_FRAME(0xFFF6, 0xFFF6, 0, 10),
```

Comment:
`AFFINEANIMCMD_FRAME(-10, -10, 0, 10), `
(Please also fix other instances in this file)